### ISSUE TITLE
IndentationError: unindent does not match any outer indentation level

### ISSUE THREAD

**Issue by DubMonstaa** [mainde/klipper-moonraker-tapo#1]
Hi!
Sorry to bother you but I'm encountering an issues in the logs which I do not understand as it is above my knowledge : D so maybe you could help.

So I bought a pair of P110 and setting them up today for them to work with your solution (I have the Home Assistant through Docker one working ok but would like to get rid of it, I am not really using HA for anything else), but I am hitting a roadblock, this one:
![image](https://github.com/mainde/klipper-moonraker-tapo/assets/126898892/aab9f539-3e37-4958-8556-475da894d7e3)

I redid the install several times, same issue. 
Any help would be appreciated!
Thank you!

**Comment by mainde** [comment]
Hiya! The error you're encountering is suggesting that the line of code in question (line 49) is not at the correct level of indentation/spacing relatively to the ones around it. Which suggests that one of us two has made a mistake while copying code to/from this github repo and the lines got misaligned.

And I think it was me, apologies! 😅 This line should have been nested just under the "elif":
![image](https://github.com/mainde/klipper-moonraker-tapo/assets/14027750/f114e35f-d076-430c-96d8-6a4464cfc8a9)

I've now corrected the main page, you can delete your `server.py` and take a fresh copy or just add three spaces at the start of the offending line.

**Comment by DubMonstaa** [comment]
Hi! Thank you for your reply!
I did as you said but unfortunatly there is the same error.. Here is my server.py:


_#!/usr/bin/python3

import json
import signal
from http.server import SimpleHTTPRequestHandler, HTTPServer
from PyP100 import PyP110

TAPO_ADDRESS = "192.168.0.18"
TAPO_USERNAME = "cristi.isvoranu@gmail.com"
TAPO_PASSWORD = "Ratri3.1"

p100 = PyP110.P110(TAPO_ADDRESS, TAPO_USERNAME, TAPO_PASSWORD)
p100.handshake()
p100.login()

running = True

def exit_gracefully(*args, **kwargs):
    print("Terminating..")
    global running
    running = False

signal.signal(signal.SIGINT, exit_gracefully)
signal.signal(signal.SIGTERM, exit_gracefully)


class MyHttpRequestHandler(SimpleHTTPRequestHandler):

      def do_GET(self):
         self.send_response(200)
         self.send_header('Content-type', 'application/json')
         self.end_headers()

         global p100  # hack to fight Tapo session expiry, somehow just redoing handhsake and login doesn't work
         try:
             if self.path == "/on":
                 p100.turnOn()
             elif self.path == "/off":
                 p100.turnOff()
             self.wfile.write(json.dumps(p100.getDeviceInfo()).encode("utf-8"))
         except Exception as e:  # YOLO
             p100 = PyP110.P110(TAPO_ADDRESS, TAPO_USERNAME, TAPO_PASSWORD)
             p100.handshake()
             p100.login()
             if self.path == "/on":
                 p100.turnOn()
             elif self.path == "/off":
                 p100.turnOff()
             self.wfile.write(json.dumps(p100.getDeviceInfo()).encode("utf-8"))
        return


if __name__ == '__main__':
    server_class = HTTPServer
    handler_class = MyHttpRequestHandler
    server_address = ('127.0.0.1', 56427)

    httpd = server_class(server_address, handler_class)
    # intentionally making it slow, it doesn't need to react quickly
    httpd.timeout = 1  # seconds

    try:
        while running:
            httpd.handle_request()
    except KeyboardInterrupt:
        pass_


I am using WinSCP to place and edit it, would that be the issue?

Thank you!

**Comment by mainde** [comment]
I think you have accidentally shared your own username and password for the Tapo 😅 you may want to delete your comment.

Anyway let me commit a file that you can download directly and there's no need to do the copy/paste

**Comment by mainde** [comment]
I've deleted your comment since it had your tapo credentials, you may still want to change your password.
Anyway there was another mistake in my file, apologies again, a missing space before "return" on the following line.

It should be good now to copy from the main page or download from here https://github.com/mainde/klipper-moonraker-tapo/blob/main/server.py

**Comment by DubMonstaa** [comment]
I sure did that 😅 I was tired and wanted to test it before going to bed, my bad... Changed the password shortly after posting the reply. 
I downloaded your file, redid the config and it is working now, eventhough that error is still present when checking with  ```journalctl -u tapo```. But now, ```service tapo status``` return an "active" status, as before it was not doing that.

But seing that it's working even with that error, I would call it a success 😄 

One other small note, I think ```systemctl enable tapi.service``` should be changed to ```systemctl enable tapo.service```, it was also throwing another error. 
![image](https://github.com/mainde/klipper-moonraker-tapo/assets/126898892/7659e1ce-0ed1-4e9b-b08d-af736d069c41)

Thank you for your help and time! Really glad it is working with your solution, the HA one is also good but it is more involved to setup and maybe a "heavier process" for the PI.
